### DISCUSSION TITLE
Meta_Hours_10

### DISCUSSION THREAD

**Discussion by yangwao** [kodadot/nft-gallery#3030] [Meta-Hours]
![creative-applications-1600-1000-1200x750](https://user-images.githubusercontent.com/5887929/168998932-84244e6b-9032-41fb-88cb-45265f797d86.png)
### Discord
- https://discord.com/events/840514076538830888/976398374217605130

### Summer of events

There are a couple of events we are attending upcoming weeks, which is why we postponed from our usual bi-weekly schedule
- BlockSplit (Split, Croatia)
- UTXO.cz (Prague, Czechia)
- DotsamaPrague (Prague, Czechia)
- Polkadot Decoded (Berlin, Germany)
- Unchained (Berlin, Germany)

Apart from the usual schedule, we will have new updates for you:)

## Basilisk aka Snek
Long outstanding Basilisk is incoming and it's already available for testing at https://beta.kodadot.xyz/bsx 
Perfect time to uncover bugs or major UX fails we have by [reporting them to our Github Issues](https://github.com/kodadot/nft-gallery/issues/new/choose) 

### What's in the Snek box? 📦
**- Royalties** - finally we can offer the first trustless royalties in the Dotsama NFT landscape. You can mint NFTs, which will be forward-compatible with other para chains and transferable across Kusama & Polkadot ecosystem. 
 <img width="553" alt="image" src="https://user-images.githubusercontent.com/5887929/177550539-90999e68-da70-457a-bd3f-2d0c0dadbad2.png">
**- Proxied Royalties** - same as above, yet you can set other than author address to receive royalties. This can open new subset of creative use cases for your DAO and someone else can mint and pass your royalties on your behalf
<img width="551" alt="image" src="https://user-images.githubusercontent.com/5887929/177552743-cd6534f6-ab4a-427e-9620-83ca63ebaf3a.png">
**- Make Offers** - the best upcoming feature is that potential buyers can on a runtime level make offers with various sizes and current owners of particular NFT can accept one offer and trustlessly transfer NFT to the author of the offer.
<img width="957" alt="image" src="https://user-images.githubusercontent.com/5887929/177556483-95415aa2-a0dd-4f12-b3db-650cdcd80cbd.png">
<img width="969" alt="image" src="https://user-images.githubusercontent.com/5887929/177559434-0c9261d4-0af7-4f13-b466-5444b2732f9a.png">

Feel free to jump into Snek channel on our Discord and drop your BSX address and we can send you testing BSX tokens, so you can experience features like **Royalties, Proxied Royalties and Make Offer.** 

- https://github.com/kodadot/nft-gallery/issues/660

## Moonsama
- We've realized the importance of collections on EVM-based parachains, like MoonRiver, MoonBeam and Astar.  That's why we've started implementing MoonSama collections into KodaDot, which would help us bring a bigger audience to KodaDot and offer them marketplace insights we already providing for RMRK community. We will keep you posted once Moonsama lands beta channel and will be ready to sneak peek.
 - https://github.com/kodadot/nft-gallery/issues/3161

## Upgrades
- Right now on the beta of basilisk we encounter a weird bug with pinning and fetching fresh mints NFTs, so once you mint the stuff it's not there. We've made a tracking issue, so once it's closed, you'll see an update on your client.  
- https://github.com/kodadot/nft-gallery/issues/3334 

### Discord updates

- now you can see most of the developer events (issue comments & PRs) happening from repositories  in channel dev-events, added these to our repositories to see activity in one place
  - snek
  - rubick
  - click
  - docs
  - packages
  - nft-gallery

### Recovering 

Hence team is still in recovery, I bet upcoming Meta Hours 11 would be much more packed https://github.com/kodadot/nft-gallery/discussions/3344

**Comment by petersopko** [comment]
I'll leave the new implementations / integrations to @vikiival and @yangwao since they've expanded on them in great detail, I'll have just three brief things to mention
- upcoming refactor
- automated tests as github actions
-  conferences!

**Comment by DeluneClaire** [comment]
Claire's minute:

- Feedback to previous conferences 
- Coordination 
- Summary of the KodaDot survey
- Summary of the DotSama Prague survey
- The importance of the K-family environment

**Comment by vikiival** [comment]
**Viki's Meta Minutes**
-

### DEV in numbers:
`is:pr is:merged created:>=2022-05-19`

> 106 PRs merged

### Basilisk:

We finished the Basilisk implementation and are polishing the last bugs before unlocking tokens.

Moreover, we got a promise from @green-jay that we will have auctions available on the testnet:
It would mean that we can sell our NFT via:
- English auction
- Dutch auction
- Top-up

### Moonsama:

Along with finishing BSX, we are participating in the Polkadot North America Hackathon.
We are researching how to bring @moonsama to the KodaDot via @subsquid EVM indexer.

<img width="1131" alt="Screenshot 2022-06-28 at 09 42 26" src="https://user-images.githubusercontent.com/22471030/177514094-235fa73c-738e-4c8b-bb2f-d02420bbe2c3.png">


### API:

During the last weeks, we have been working on a new set of packages, internally called `uniquery` that should bring the possibility to external projects to leverage our API in their dApps without previous knowledge of GraphQL.

The goal is basically simple:

```ts
const query = SubQuery.getNftById(id)
```

### Events:

As yangwao previously mentioned, we traveled a lot to spread the word about KodaDot and also participated in various hackathons:


#### BlockSplit x Encode:
With @petersopko, we hacked a possibility to generate a title and description of the NFT just from the picture. It is done via a pipeline of AI, namely Azure Vision and GPT-3


[Try it on Beta KodaDot](https://beta.kodadot.xyz/rmrk/creative)


#### ETHPrague:
With the help of @frankbevr, we hacked a simple game similar to Tamagotchi where you need to consistently water a plant, and thanks to the mathematics, the tree grows.
This is important as we are researching how to bring generative art into the KodaDot.

[See vikiival/tamago implemetation](https://github.com/vikiival/tamago)


### Towards reactivity:

We implemented a package for conditional rendering. It was already presented at the Blokchaton Workshop and should be integrated into KodaDot's sandbox soon.

**Comment by damskyftw** [comment]
My chilled out entry:

- substack up and running
- Getting back on track after conferences run
- Dotsama Prague maintaining
- Medium articles will be finally going out on fixed schedule (Thursday or monday)
- bsx tutorial
